modules/invoicein/model/invoicein.class.php

```diff
@@ -400,11 +400,36 @@ public function journal() {
                     $VoucherH['voucher_AutomaticReason']    = "Fra innk faktura ID: " . $InvoiceO->ID;
 
                     $VoucherH['voucher_KID']                = $InvoiceO->KID;
-                    $VoucherH['voucher_DepartmentID']       = $InvoiceO->Department;
-                    $VoucherH['voucher_ProjectID']          = $InvoiceO->Project;
                     $VoucherH['voucher_InvoiceID']          = $InvoiceO->InvoiceNumber;
                     $VoucherH['voucher_AccountPlanID']      = $InvoiceO->SupplierAccountPlanID;
 
+                    $invoice_supplier_accountplan           = accounting::get_accountplan_object($VoucherH['voucher_AccountPlanID']);
+                    if (!empty($invoice_supplier_accountplan)) {
+                          $invoice_supplier_motkonto_accountplan = accounting::get_accountplan_object($invoice_supplier_accountplan->MotkontoResultat1);
+                    }
+                    if (!empty($invoice_supplier_motkonto_accountplan)) {
+                      if ($invoice_supplier_motkonto_accountplan->EnableDepartment == 1 && !is_null($invoice_supplier_motkonto_accountplan->DepartmentID)) {
+                        $invoice_supplier_motkonto_default_department = $invoice_supplier_motkonto_accountplan->DepartmentID;
+                      } else {
+                        $invoice_supplier_motkonto_default_department = NULL;
+                      }
+                      if ($invoice_supplier_motkonto_accountplan->EnableProject == 1 && !is_null($invoice_supplier_motkonto_accountplan->ProjectID)) {
+                        $invoice_supplier_motkonto_default_project = $invoice_supplier_motkonto_accountplan->ProjectID;
+                      } else {
+                        $invoice_supplier_motkonto_default_project = NULL;
+                      }
+                    }
+                    if($InvoiceO->Department) {
+                        $VoucherH['voucher_DepartmentID'] = $InvoiceO->Department;
+                    } else {
+                        $VoucherH['voucher_DepartmentID'] = $invoice_supplier_motkonto_default_department;
+                    }
+                    if($InvoiceO->Project) {
+                        $VoucherH['voucher_ProjectID'] = $InvoiceO->Project;
+                    } else {
+                        $VoucherH['voucher_ProjectID'] = $invoice_supplier_motkonto_default_project;
+                    }
+
```

Comment:
Only assign default directly to the VoucherH['voucher_Dep/Proj'] if not set in $invoice->Dep/Proj